cocos/2d/CCParticleSystem.cpp

```diff
@@ -674,158 +841,41 @@ void ParticleSystem::update(float dt)
-    }
-    else if (_positionType == PositionType::RELATIVE)
-    {
-        currentPosition = _position;
-    }
-
+    
     {
-        Mat4 worldToNodeTM = getWorldToNodeTransform();
+        for (int i = 0; i < _particleCount; ++i)
+        {
+            _particleData.timeToLive[i] -= dt;
+        }
         
-        while (_particleIdx < _particleCount)
+        for (int i = 0; i < _particleCount; ++i)
         {
-            tParticle *p = &_particles[_particleIdx];
-
-            // life
-            p->timeToLive -= dt;
-
-            if (p->timeToLive > 0) 
-            {
-                // Mode A: gravity, direction, tangential accel & radial accel
-                if (_emitterMode == Mode::GRAVITY)
-                {
-                    Vec2 tmp, radial, tangential;
-
-                    // radial acceleration
-                    if (p->pos.x || p->pos.y)
-                    {
-                        radial = p->pos.getNormalized();
-                    }
-                    tangential = radial;
-                    radial = radial * p->modeA.radialAccel;
-
-                    // tangential acceleration
-                    float newy = tangential.x;
-                    tangential.x = -tangential.y;
-                    tangential.y = newy;
-                    tangential = tangential * p->modeA.tangentialAccel;
-
-                    // (gravity + radial + tangential) * dt
-                    tmp = radial + tangential + modeA.gravity;
-                    tmp = tmp * dt;
-                    p->modeA.dir = p->modeA.dir + tmp;
-                    
-                    // this is cocos2d-x v3.0
-//                    if (_configName.length()>0 && _yCoordFlipped != -1)
-
-                    // this is cocos2d-x v3.0
-                    tmp = p->modeA.dir * dt * _yCoordFlipped;
-                    p->pos = p->pos + tmp;
-                }
-
-                // Mode B: radius movement
-                else 
-                {                
-                    // Update the angle and radius of the particle.
-                    p->modeB.angle += p->modeB.degreesPerSecond * dt;
-                    p->modeB.radius += p->modeB.deltaRadius * dt;
-
-                    p->pos.x = - cosf(p->modeB.angle) * p->modeB.radius;
-                    p->pos.y = - sinf(p->modeB.angle) * p->modeB.radius;
-                    p->pos.y *= _yCoordFlipped;
-				}
-
-                // color
-                p->color.r += (p->deltaColor.r * dt);
-                p->color.g += (p->deltaColor.g * dt);
-                p->color.b += (p->deltaColor.b * dt);
-                p->color.a += (p->deltaColor.a * dt);
-
-                // size
-                p->size += (p->deltaSize * dt);
-                p->size = MAX( 0, p->size );
-
-                // angle
-                p->rotation += (p->deltaRotation * dt);
-
-                //
-                // update values in quad
-                //
-
-                Vec2    newPos;
-
```

Comment:
These code are missed, please confirm that whether it is should be deleted or not.